### ISSUE TITLE
Create YOLO loss function

### ISSUE THREAD

**Issue by jpsember** [jpsember/java-ml#31]
Relevant:  https://neptune.ai/blog/pytorch-loss-functions

**Comment by jpsember** [comment]
Why are the IOU calculations suspect?
```
true_box_min
[        │        │        │        │        │        ]
[        │        │ -0.188 │  0.306 │        │        ]
[        │        │        │        │        │        ]
true_box_max
[       │       │       │       │       │       ]
[       │       │  0.69 │  1.01 │       │       ]
[       │       │       │       │       │       ]
pred_box_min
[       │       │       │       │       │       ]
[       │       │ -0.48 │  1.45 │       │       ]
[       │       │       │       │       │       ]
pred_box_max
[       │       │       │       │       │       ]
[       │       │  1.34 │ -0.29 │       │       ]
[       │       │       │       │       │       ]

```
the (pred) box max seems to be less than the box min.

**Comment by jpsember** [comment]
The predicted w,h seems to be negative sometimes!  We should use a function that maps -inf...+inf to 0..+inf

**Comment by jpsember** [comment]
The true and predicted width and height look reasonable 👍 
```
true_wh
[       │       │  1.90 │  2.20 ]
pred_wh
[       │       │  1.71 │  1.95 ]
wh true-pred
[         │         │  0.0008 │  0.0002 ]
```
but they don't match when converted to Scredit scripts
